app/models/host/managed_extensions.rb

```diff
@@ -41,16 +41,23 @@ def delReboot
     # no reboot on orchestration rollback
   end
 
-  def setKexec
+  def render_kexec_template
     template = provisioning_template(:kind => 'kexec')
     raise ::Foreman::Exception.new(N_("Kexec template not associated with operating system")) unless template
-    @host = self
-    @kexec_kernel, @kexec_initrd = operatingsystem.boot_files_uri(@host.medium, @host.architecture, @host)
-    # try to parse JSON and error out early
-    json = JSON.parse(unattended_render(template))
+
+    source = Foreman::Renderer::Source::String.new(name: template.name, content: template.template)
+    scope = Foreman::Renderer.get_scope(host: self, source: source)
+    json = JSON.parse(Foreman::Renderer.render(source, scope))
     ::Foreman::Exception.new(N_("Kernel kexec URL is invalid: '%s'"), json['kernel']) unless json['kernel'] =~ /\Ahttp.+\Z/
     ::Foreman::Exception.new(N_("Init RAM kexec URL is invalid: '%s'"), json['initrd']) unless json['initrd'] =~ /\Ahttp.+\Z/
-    old.becomes(Host::Discovered).kexec json.to_json
+    json.to_json
+  #rescue StandardError => e
```

Comment:
What do we do with this commented stuff?

Reply:
Damn a leftover from debugging. Removing.